docs/TEMPORARY_DIRECTORIES.md

```diff
@@ -0,0 +1,201 @@
+   long as you have the file descriptor open you can still access the file for
+   both reading and writing. When operating this way it is recommended to delete
+   the files right after creating them to ensure that on abnormal program
+   termination the files or directories are implicitly released by the kernel.
+
+2. 🥇 Use `memfd_create()` or `O_TMPFILE`. This is an extension of the suggestion
+   above: files created this way are never linked under a filename in the file
+   system. This means they are not subject to ageing (as they come unlinked out
+   of the box), and there's no time window where the file exists but not yet
+   deleted, and thus behaviour is fully robust towards abnormal program
+   termination as there are never files on disk that need to be explicitly
+   deleted.
+
+3. 🥇 Operate below a sub-directory of `/tmp/` and `/var/tmp/` you created, and
+   take a BSD file lock ([`flock(dir_fd,
+   LOCK_SH)`](http://man7.org/linux/man-pages/man2/flock.2.html)) on that
+   sub-directory. This is particularly interesting when operating on more than a
+   single file, or on file nodes that are not plain regular files, for example
+   when extracting a tarball to a temporary directory. The ageing algorithm
+   will skip all directories (and everything below them) that are locked
+   through a BSD file lock. As BSD file locks are automatically released when
+   the file descriptor they are taken on is closed, and all file descriptors
+   opened by a process are implicitly closed when it exits this is a robust
+   mechanism that ensures all generated temporary files are subject to ageing
+   when the program that owns them abnormally dies, but not while it is still
+   running. This is a particularly interesting option when decompressing
+   tarballs that contain files with old modification/access times, as extracted
+   files are otherwise immediately candidates for deletion by the ageing
+   algorithm. The [`flock`](http://man7.org/linux/man-pages/man1/flock.1.html)
+   tool of the `util-linux` packages makes this concept available to shell
+   scripts. Note that `systemd-tmpfiles` only checks for BSD file locks on
+   directories, locks on other types of file nodes (including regular files)
+   are not considered.
+
+4. Keep the access time of all temporary files created current. In regular
+   intervals, use `utimensat()` or a related call to update the access time
+   ("atime") of all files that shall be kept around. Since the ageing algorithm
+   looks at the access time of files when deciding whether to delete them, it's
+   sufficient to update their access times in sufficiently frequent intervals to
+   ensure the files are not deleted. Since most applications (and tools such as
+   `ls`) primarily care for the modification time (rather than the access time)
+   using the access time for this purpose should be acceptable.
+
+5. Set the "sticky" bit on regular files. The ageing logic skips deletion of
```

Comment:
Didn't this get dropped in c6878637502b1717a110a9a7e8bba32a8583fcdf?

Reply:
Nope, it's still there:

https://github.com/systemd/systemd/blob/master/src/tmpfiles/tmpfiles.c#L648

The ageing stuff doesn't use rm_rf()